### PR TITLE
[Doc] Add NNStreamer wiki to github.io

### PR DESCRIPTION
**Pull request by gichan-jang**
To provide more information to the user, add part of the NNStreamer wiki to nnstreamer.github.io.
A wiki page is a temporary archive of the NNStreamer documents that change frequently or are being written.
"stable" documents from nnstreamer wiki is migrated to /Documentations/.

Migrated docs:
 - Usage examples and screenshots
 - Example: GStreamer launch scripts
 - Synchronization Policies at Mux and Merge
 - AD: Data Type, Caps, and Flow Design
 - Rank counting with other/tensor types

I will remove the wiki pages after this PR is merged.

~~Discussion:
But hyperlink rule of the github.wiki and hotdoc is not compatible.
So, one of the two has a broken link.
In my opinion, it seems good to unify the document into github.io.
It is freely modified in the wiki without PR and the contents are automatically reflected in github.io.~~

Signed-off-by: Gichan Jang <gichan2.jang@samsung.com>

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [*]Skipped

### CODE REVIEW

File: Documentation/data-type-and-flow-control.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+---
+title: Data type and Flow cntrol
```
Comment (jaeyun-jung) [inline Documentation/data-type-and-flow-control.md:2]: typo cntrol

File: Documentation/data-type-and-flow-control.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+Typestring = (string) { float32, float64, int64, uint64, int32, uint32, int16, uint16, int8, uint8 }
+Dimensions = (string) Dimension
+           | (string) Dimension, Dimensions
+Dimension = (string) [1-65535]:[1-65535]:[1-65535]:[1-65535]
+```
+
+The buffer of ```other/tensors``` streams have multiple memory chunks. Each memory chunk represents one single tensor in the buffer format of ```other/tensor```. With default configurations of Gstreamer 1.0, the maximum allowed number of memory chunks in a buffer is **16**. Thus, with such configurations of Gstreamer 1.0, ```other/tensors``` may include up to **16** ```other/tensor```.
+
+
+## other/tensorsave
```
Comment (jaeyun-jung) [inline Documentation/data-type-and-flow-control.md:79]: Discussion: `o/tensorsave` is necessary?

File: Documentation/data-type-and-flow-control.md (hunk 1 above)
Comment (gichan-jang) [reply Documentation/data-type-and-flow-control.md]: Updated.

File: Documentation/data-type-and-flow-control.md (hunk 2 above)
Comment (gichan-jang) [reply Documentation/data-type-and-flow-control.md]: `other/tensorsave` has not been implemented. Marked as `TBU`

File: Documentation/how-to-run-examples.md [hunk 3]
```diff
@@ -2,6 +2,288 @@
...
+* https://github.com/nnstreamer/nnstreamer/blob/master/Documentation/getting-started.md
+
+- Install related packages before building nnstreamer and examples.
+1. ninja-build, meson (>=0.42)
+2. liborc (>=0.4.25, optional)
+3. tensorflow, protobuf (>=3.4.0)
+4. tensorflow-lite
+
+- Build options (meson)
+1. enable-tensorflow : default true, add option to disable tensorflow (-Denable-tensorflow=false)
```
Comment (jaeyun-jung) [inline Documentation/how-to-run-examples.md:54]: Let's update these build options.

Comment (gichan-jang) [reply Documentation/how-to-run-examples.md]: Updated build option and broken link.

File: Documentation/data-type-and-flow-control.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+# Flow Control
+
+## Timestamps
+
+We have the following principles for timestamp policies. Timestamping policies of ```tensor_*``` filters should follow the given principles.
+
+- Timestamp from the input source (sensors) should be preserved to sink elements.
+- When there are multiple flows merging into one (an element with multiple sink pads), a timestamp of the most recent should be preserved.
+    - For the current frame buffer in a sink pad, ```i``` in ```1 ... n```, ```FB(i)```, and ```T(FB(i))``` is the timestamp of the current frame buffer, the timestamp of the corresponding frame buffer at source pads generated by ```FB(1)``` ... ```FB(n)``` is ```max(i = 1 .. n, T(FB(i)))```, where larger timestamp value means the more recent event.
+    - Example: when multiple frames of the same stream are aggregated by ```tensor_aggregator```, according to this principle, the timestamp of the last frame of the aggregated frames is used for output.
```
Comment (jaeyun-jung) [inline Documentation/data-type-and-flow-control.md:96]: need update: tensor-aggregator does not have multi pads, and timestamp for outging buffer is not the last frame's.
timestamp of first buffer (+duration)

this example needs update (IMO, tensor-mux is better)

Comment (gichan-jang) [reply Documentation/data-type-and-flow-control.md]: Right. It's about `tensor_mux`. 
Modified related descriptions.

File: Documentation/data-type-and-flow-control.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+# GStreamer data types (pad capabilities)
+
+All NNStreamer's GStreamer data types as pad capabilities (```other/tensor*```) have the following common rules
+
+1. In each buffer, there is only ONE frame. That is for each buffer, with the type of ```other/tensor```, there is only one instance of tensor for each buffer at any time. There cannot be multiple tensors in each buffer.
+2. The data types do not hold data semantics. Filters should NOT try to determine data semantics (e.g., is it a video?) dynamically based solely on the dimensions, framerates, or element types of the data types. However, if a filter has additional information available including property values from pipeline developers or users, a filter may determine data semantics. For example, ```tensor_decoder``` transforms ```other/tensor``` stream into ```video/x-raw``` or ```text/x-raw``` depending on the property values.
+
+## other/tensor
+
+The GST pad capability has the following structure:
```
Comment (jaeyun-jung) [inline Documentation/data-type-and-flow-control.md:16]: The GStreamer pad capability has ...
compated to line 60 (The GStreamer pad capability of ```other/tensors``` is as follows:)

File: Documentation/how-to-run-examples.md [hunk 6]
```diff
@@ -2,6 +2,285 @@
...
+$ cd /usr/lib/nnstreamer/bin
+```
+
+As of 2018/10/13, we support 16.04 and 18.04
+
+## Build examples (Ubuntu 16.04 and 18.04)
+* https://github.com/nnstreamer/nnstreamer/blob/master/Documentation/getting-started.md
+
+- Install related packages before building nnstreamer and examples.
+1. ninja-build, meson (>=0.42)
```
Comment (jaeyun-jung) [inline Documentation/how-to-run-examples.md:48]: meson (>=0.50)

File: Documentation/how-to-run-examples.md [hunk 7]
```diff
@@ -2,6 +2,285 @@
...
+2. liborc (>=0.4.25, optional)
+3. tensorflow, protobuf (>=3.4.0)
```
Comment (jaeyun-jung) [inline Documentation/how-to-run-examples.md:50]: protobuf 3.6.0

File: Documentation/how-to-run-examples.md [hunk 8]
```diff
@@ -2,6 +2,285 @@
...
+```
+
+- Download tensorflow-lite model & labels.
+```
+$ cd $NNST_ROOT/bin/
+$ mkdir tflite_model       # for the example image classification
+$ cd tflite_model
+$ wget https://github.com/nnsuite/testcases/raw/master/DeepLearningModels/tensorflow-lite/Mobilenet_v1_1.0_224_quant/mobilenet_v1_1.0_224_quant.tflite
+$ wget https://github.com/nnsuite/testcases/raw/master/DeepLearningModels/tensorflow-lite/Mobilenet_v1_1.0_224_quant/labels.txt
+$ cd ..
```
Comment (jaeyun-jung) [inline Documentation/how-to-run-examples.md:106]: plz check example repo, tflite-object detection example.

remove line 106 ~ 108
now tflite_model directory is default.

Comment (jaeyun-jung) [reply Documentation/how-to-run-examples.md:106]: It is better to update this guide to use 'get-model.sh' in example repo.

File: Documentation/data-type-and-flow-control.md (hunk 5 above)
Comment (gichan-jang) [reply Documentation/data-type-and-flow-control.md]: Nice! updated.

File: Documentation/how-to-run-examples.md (hunk 6 above)
Comment (gichan-jang) [reply Documentation/how-to-run-examples.md]: updated.

File: Documentation/how-to-run-examples.md (hunk 8 above)
Comment (gichan-jang) [reply Documentation/how-to-run-examples.md]: Changed to use get-mode.sh. Thanks.

File: Documentation/synchronization-policies-at-mux-merge.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+# Synchronization policies at Mux and Merge
+There are multiple synchronization policies for tensor_mux and tensor_merge.
+They are based on PTS ( presentation timestamp in nanoseconds (as a GstClockTime) ) and assume that every tensor buffer has PTS and can be accessed by GST_BUFFER_PTS(buf).
+
+However, there is stream which does not have PTS such as application/octet-stream. In such cases, tensor_converter generates timestamp and set PTS. If framerate is given, tensor_converter generates proper PTS according to framerate and for the case without framerate, PTS is decided with running time which is calculated as absolute-time - base-time.
+
+Currently, three synchronization policies are implemented. 
+
+# No synchronization : default
```
Comment (jaeyun-jung) [inline Documentation/synchronization-policies-at-mux-merge.md:13]: remove 'default' 
mux slowest / merge no-sync

Comment (gichan-jang) [reply Documentation/synchronization-policies-at-mux-merge.md]: Right. I updated more description about the sync option examples. PTAL.

**Review by leemgs — APPROVED** [review]
Acked-by: Geunsik Lim.

**Pull request merged**